includes/wikia/api/SendGridPostBackApiV3Controller.class.php

```diff
@@ -24,6 +24,13 @@ public function log() {
 		if (empty($events)) {
 			Wikia::log(__METHOD__, false, "No data to process", true);
 			wfProfileOut( __METHOD__ );
+			return;
+		}
+
+		if (!is_array($events)) {
+			Wikia::log(__METHOD__, false, "Supplied data is not an array (JSON begins with): " . @substr($data,0,100), true);
+			wfProfileOut( __METHOD__ );
+			return;
```

Comment:
As we're here why not replace ``Wikia::log`` calls with `WikiaLogger` (via `Loggable` trait) and log with context.